@Test
public
void
checkThatUnresolvedBugTestFail
(
)
throws
InitializationError
{
JUnitCore
junit
=
new
JUnitCore
(
)
;
FailureListener
failures
=
new
FailureListener
(
)
;
junit
.
addListener
(
failures
)
;
for
(
CtMethod
unresolvedTest
:
testMethods
)
{
Class
testClass
=
unresolvedTest
.
getDeclaringType
(
)
.
getActualClass
(
)
;
String
testMethod
=
unresolvedTest
.
getSimpleName
(
)
;
junit
.
run
(
new
RunDespiteIgnore
(
testClass
,
testMethod
)
)
;
assertTrue
(
failures
.
failures
.
contains
(
testMethod
+
"("
+
unresolvedTest
.
getDeclaringType
(
)
.
getQualifiedName
(
)
+
")"
)
)
;
System
.
out
.
println
(
unresolvedTest
.
getDeclaringType
(
)
.
getQualifiedName
(
)
+
"#"
+
testMethod
+
" fails as expected"
)
;
}
}
